File: back/src/main/java/com/back/domain/trend/snapshot/entity/DailyKeywordCount.java
package com.back.domain.trend.snapshot.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "daily_keyword_count",
        uniqueConstraints = @UniqueConstraint(columnNames ={"date","keyword"}))
public class DailyKeywordCount {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDate date;
    private String keyword;
    private long frequency;

    public DailyKeywordCount(LocalDate date, String keyword, long frequency) {
        this.date = date;
        this.keyword = keyword;
        this.frequency = frequency;
    }

    public void updateFrequency(long frequency) {
        this.frequency = frequency;
    }
}

File: back/src/main/kotlin/com/back/domain/trend/snapshot/entity/DailyKeywordCount.kt
package com.back.domain.trend.snapshot.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import java.time.LocalDate

@Entity
@Table(
    name = "daily_keyword_count",
    uniqueConstraints = [UniqueConstraint(columnNames = ["date", "keyword"])]
)
class DailyKeywordCount(
    date: LocalDate,
    keyword: String,
    frequency: Long
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        protected set

    var date: LocalDate = date
        protected set

    var keyword: String = keyword
        protected set

    var frequency: Long = frequency
        protected set

    fun updateFrequency(frequency: Long) {
        this.frequency = frequency
    }
}
